sdk/iot/azure-iot-modelsrepository/azure/iot/modelsrepository/client.py

```diff
@@ -0,0 +1,173 @@
+        )
+        # api_version = _DEFAULT_API_VERSION if api_version is None else api_version
+
+        kwargs.setdefault("api_verison", api_version)
+
+        # NOTE: depending on how this class develops over time, may need to adjust relationship
+        # between some of these objects
+        self.fetcher = _create_fetcher(
+            location=repository_location, api_version=api_version, **kwargs
+        )
+        self.resolver = resolver.DtmiResolver(self.fetcher)
+        self.pseudo_parser = pseudo_parser.PseudoParser(self.resolver)
+
+    def get_models(self, dtmis, dependency_resolution=DEPENDENCY_MODE_DISABLED):
+        """Retrieve a model from the Models Repository.
+
+        :param list[str]: The DTMIs for the models you wish to retrieve
+        :param str dependency_resolution : Dependency resolution mode. Possible values:
+            - "disabled": Do not resolve model dependencies
+            - "enabled": Resolve model dependencies from the repository
+            - "tryFromExpanded": Attempt to resolve model and dependencies from an expanded DTDL
+                    document in the repository. If this is not successful, will fall back on
+                    manually resolving dependencies in the repository
+
+        :raises: ValueError if given an invalid dependency resolution mode
+        :raises: ResolverError if there is an error retreiving a model
+
+        :returns: Dictionary mapping DTMIs to models
+        :rtype: dict
+        """
+        if dependency_resolution == DEPENDENCY_MODE_DISABLED:
+            model_map = self.resolver.resolve(dtmis)
+        elif dependency_resolution == DEPENDENCY_MODE_ENABLED:
+            # Manually resolve dependencies using pseudo-parser
+            base_model_map = model_map = self.resolver.resolve(dtmis)
+            base_model_list = [model for model in base_model_map.values()]
+            model_map = self.pseudo_parser.expand(base_model_list)
+        elif dependency_resolution == DEPENDENCY_MODE_TRY_FROM_EXPANDED:
+            # Try to use an expanded DTDL to resolve dependencies
+            try:
+                model_map = self.resolver.resolve(dtmis, expanded_model=True)
+            except resolver.ResolverError:
+                # Fallback to manual dependency resolution
+                base_model_map = model_map = self.resolver.resolve(dtmis)
+                base_model_list = [model for model in base_model_map.items()]
+                model_map = self.pseudo_parser.expand(base_model_list)
+        else:
+            raise ValueError("Invalid dependency resolution mode: {}".format(dependency_resolution))
+        return model_map
+
+
+class ModelsRepositoryClientConfiguration(Configuration):
+    """ModelsRepositoryClient-specific variant of the Azure Core Configuration for Pipelines"""
+
+    def __init__(self, **kwargs):
+        super(ModelsRepositoryClientConfiguration, self).__init__(**kwargs)
```

Comment:
Would be good to itemize in docstrings kwargs that can be passed in.

Reply:
That was my first thought, however it seems fairly impractical given how kwargs work in the Track 2 context - all kwargs are passed to all internal objects, so kwargs are the superset of all options related to not just the internal pipeline, but all policies, etc. I suspect this is why no other python clients seem to be listing them either. This is something I will raise during review

Reply:
Sounds good